api/src/migrations/main/1721691825980-project-plan-table.js

```diff
@@ -0,0 +1,79 @@
+const { MigrationInterface, QueryRunner } = require("typeorm");
+
+module.exports = class ProjectPlanTable1721691825980 {
+
+    async up(queryRunner) {
+        console.log('Starting new project_plan_code migration.');
+        await queryRunner.query(`
+        -- add new table - project_plan_code
+            drop table if exists app_fom.project_plan_code;  
+            create table if not exists app_fom.project_plan_code  
+            ( 
+                code varchar(15) not null primary key,
+                description varchar not null,  
+                revision_count integer not null default 1,
+                create_timestamp timestamptz not null default now(),  
+                create_user varchar not null,  
+                update_timestamp timestamptz,  
+                update_user varchar  
+            );
+
+        -- comments on columns.
+            comment on table  app_fom.project_plan_code is 'Categories to classify FOM submission from different licensee plan';
+            comment on column app_fom.project_plan_code.code is 'Code value';
+            comment on column app_fom.project_plan_code.description is 'Code description';
+            comment on column app_fom.project_plan_code.create_timestamp is 'Time of creation of the record.';
+            comment on column app_fom.project_plan_code.create_user is 'The user id who created the record.';
+            comment on column app_fom.project_plan_code.update_timestamp is 'Time of most recent update to the record.';
+            comment on column app_fom.project_plan_code.update_user is 'The user id who last updated the record.';
+            comment on column app_fom.project_plan_code.revision_count is 'Standard column for optimistic locking on updates.';
+
+        -- add code values.
+            INSERT INTO app_fom.project_plan_code(code, description, create_user) VALUES
+            ('FSP', 'Forest Stewardship Plan', CURRENT_USER), 
+            ('WOODLOT', 'Woodlot License Plan', CURRENT_USER); 
+
+        `);
+
+        console.log('Starting new columns project.project_plan_code, project.woodlot_license_number migration.');
+        await queryRunner.query(`
+        -- add new column: app_fom.project.project_plan_code column
+            ALTER TABLE app_fom.project ADD COLUMN project_plan_code varchar NOT NULL DEFAULT 'FSP' references app_fom.project_plan_code(code);
```

Comment:
Hi Ian, just for myself to understand, will this DEFAULT statement add "FSP" to all of our existing records?

Reply:
Yes